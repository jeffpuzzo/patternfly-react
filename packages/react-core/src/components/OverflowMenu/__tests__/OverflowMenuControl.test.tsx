import * as React from 'react';

import { render, screen } from '@testing-library/react';

import styles from '@patternfly/react-styles/css/components/OverflowMenu/overflow-menu';
import { OverflowMenuControl } from '../OverflowMenuControl';
import { OverflowMenuContext } from '../OverflowMenuContext';

describe('OverflowMenuControl', () => {
  test('basic', () => {
    render(
      <OverflowMenuContext.Provider value={{ isBelowBreakpoint: true }}>
        <OverflowMenuControl data-testid="test-id" />
      </OverflowMenuContext.Provider>
    );
    expect(screen.getByTestId('test-id').className).toContain(styles.overflowMenuControl);
  });

  test('Additional Options', () => {
    render(<OverflowMenuControl hasAdditionalOptions data-testid="test-id" />);
    expect(screen.getByTestId('test-id').outerHTML).toMatchSnapshot();
  });
});
