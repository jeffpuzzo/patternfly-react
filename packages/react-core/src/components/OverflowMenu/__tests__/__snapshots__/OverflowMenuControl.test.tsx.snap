// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverflowMenuControl Additional Options 1`] = `"<div class=\\"pf-c-overflow-menu__control\\" data-testid=\\"test-id\\">  </div>"`;
